### PR TITLE
RFC - Dashboard Task Queue w/ Shoryuken and ActiveJob

### PR DESCRIPTION
**Pull request by trevorb**
NOTE: This is no where close to being ready to merge, I was hoping to get some general feedback on this approach.

This is a really rough changeset that is a simple approach for a dashboard async task queue.  
### ActiveJob

http://edgeguides.rubyonrails.org/active_job_basics.html

ActiveJob is the job declaration framework that is bundled with Rails.  It's pretty standard but does some nice things like automatic serialization/deserialization of ActiveRecord objects.
### Shoryuken

https://github.com/phstc/shoryuken

Shoryuken handles pulling messages out of the queue and running them. It's mainly designed to be run from the command line but I'm having it run as part of the dashboard server.  
### Trade offs

So there are a few issues with the way I'm currently doing things.
1.   The downside of using ActiveJob is that it requires rails, so we wouldn't be able to use it with pegasus.  However, since all of these queued tasks are going to require their specific application's environment we're going to need to separate their task queue pipelines anyways.  It's just unfortunately they would be different frameworks. 
2.  Ideally we would be able to scale our task workers separately from our front-end servers but  attaching them to dashboard servers does not allow that.  However, it does simplify a lot of things like: 
   - server management (deploying new code, provisioning new servers, monitoring, etc)
   - Production vs Dev environments can be exactly the same

Given the time constraints, I think these compromises are pretty reasonable.  What do you guys think? @philbogle @wjordan @mehalshah

### CODE REVIEW

**Comment by philbogle** [comment]
I'll read and think about this, thanks.

**Comment by philbogle** [comment]
Do you want us to have the design discussion here in this PR? An alternative would be Google Docs.

**Comment by trevorb** [comment]
Lets use a Google doc:
https://docs.google.com/a/code.org/document/d/1MXm4_S7cbZH8u65fUgS0ThtxZ9KTiqXXzCk9K94bF3E/edit?usp=sharing

File: Gemfile [hunk 1]
```diff
@@ -174,3 +174,4 @@ gem 'net-scp'
 gem 'httparty'
 gem 'jquery-cookie-rails'
 gem 'oj'
+gem 'shoryuken', github: 'trevorb/shoryuken'
```
Comment (trevorb) [inline Gemfile:177]: I'll move this into the code-dot-org github repo

**Pull request closed without merging**